File: VB/E5053/Reports/ProductsReport.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.Linq
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports E5053.Models

Namespace E5053.Reports
    Partial Public Class ProductsReport
        Inherits DevExpress.XtraReports.UI.XtraReport
        Private nWindContext As New NWINDEntities()

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub ProductsReport_DataSourceDemanded(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.DataSourceDemanded
            Dim categoryID As Integer = CInt(Fix(Parameters("CategoryID").Value))
            DataSource = From product In nWindContext.Products Where product.CategoryID = categoryID Select product
        End Sub

    End Class
End Namespace

File: CS/E5053/Reports/ProductsReport.cs
using System;
using System.Drawing;
using System.Collections;
using System.Linq;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using E5053.Models;

namespace E5053.Reports {
    public partial class ProductsReport : DevExpress.XtraReports.UI.XtraReport {
        NWINDEntities nWindContext = new NWINDEntities();

        public ProductsReport() {
            InitializeComponent();
        }

        private void ProductsReport_DataSourceDemanded(object sender, EventArgs e) {
            int categoryID = (int)Parameters["CategoryID"].Value;
            DataSource = from product in nWindContext.Products
                         where product.CategoryID == categoryID
                         select product;
        }

    }
}
